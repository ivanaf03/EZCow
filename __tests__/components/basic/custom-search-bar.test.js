import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import CustomSearchBar from "../../../components/basic/custom-search-bar";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
    FontAwesomeIcon: () => null
}));

describe("CustomSearchBar", () => {

    const mockOnChangeText = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(
            <CustomSearchBar
                text={'test'}
                value={'test'}
                onChangeText={mockOnChangeText}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it("should call onChangeText when typing", () => {
        const tree = render(
            <CustomSearchBar
                text={'test'}
                value={'test'}
                onChangeText={mockOnChangeText}
            />
        );
        const input = tree.getByPlaceholderText("Buscar...");
        fireEvent.changeText(input, 'Test');
        expect(mockOnChangeText).toHaveBeenCalledTimes(1);
    });

});