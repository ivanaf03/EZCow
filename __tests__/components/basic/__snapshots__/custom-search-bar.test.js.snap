// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSearchBar should render correctly 1`] = `
<View
  style={
    [
      {
        "marginBottom": 8,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "fontFamily": "Nunito-Light",
        },
      ]
    }
  >
    test
    :
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "backgroundColor": "#ffffff",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#2254ba",
          "borderLeftColor": "#2254ba",
          "borderRightColor": "#2254ba",
          "borderTopColor": "#2254ba",
        },
        {
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Buscar..."
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "marginLeft": 8,
          },
          {
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
      value="test"
    />
    <View
      style={
        [
          {
            "height": 40,
          },
          {
            "width": 40,
          },
          {
            "alignItems": "center",
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    />
  </View>
</View>
`;
